# Traitement des dataframes et nettoyage des données

In [172]:
import pandas as pd
import matplotlib.pyplot as plt
import seaborn as sns

# Dataset 2019

In [13]:
data_2019 = pd.read_csv('../raw_data/ped_data_2019.csv')

In [14]:
data_2019

Time  O'Connell St Outside Pennys  \
0      01-01-2019 00:00:00                          244   
1      01-01-2019 00:15:00                          454   
2      01-01-2019 00:30:00                          391   
3      01-01-2019 00:45:00                          415   
4      01-01-2019 01:00:00                          319   
...                    ...                          ...   
35031  31-12-2019 22:45:00                          389   
35032  31-12-2019 23:00:00                          324   
35033  31-12-2019 23:15:00                          306   
35034  31-12-2019 23:30:00                          249   
35035  31-12-2019 23:45:00                          302   

       O'Connell St Outside Clerys  Mary Street  Capel Street  Aston Quay  \
0                           1914.0           20            16         NaN   
1                              NaN           24            62         NaN   
2                              NaN           69            99         NaN   
3                              NaN           50            61         NaN   
4                            885.0           24            47         NaN   
...                            ...          ...           ...         ...   
35031                          NaN           13            38       797.0   
35032                       1967.0           18            39       914.0   
35033                          NaN           19            69       724.0   
35034                          NaN           11            69       971.0   
35035                          NaN           14            52       513.0   

       Grafton Street @ CompuB  Talbot Street North  \
0                          NaN                  NaN   
1                          NaN                  NaN   
2                          NaN                  NaN   
3                          NaN                  NaN   
4                          NaN                  NaN   
...                        ...                  ...   
35031                    782.0                 52.0   
35032                    743.0                 48.0   
35033                    821.0                 38.0   
35034                    795.0                 69.0   
35035                    556.0                 47.0   

       Doilier Street, Burgh Quay  Dawson Street Replacement  ...  \
0                             NaN                        NaN  ...   
1                             NaN                        NaN  ...   
2                             NaN                        NaN  ...   
3                             NaN                        NaN  ...   
4                             NaN                        NaN  ...   
...                           ...                        ...  ...   
35031                       173.0                        6.0  ...   
35032                       220.0                        1.0  ...   
35033                       241.0                        4.0  ...   
35034                       357.0                        5.0  ...   
35035                       339.0                        1.0  ...   

       College Green, Bank Of Ireland  Henry Street  Westmoreland Street East  \
0                                 NaN          95.0                    1670.0   
1                                 NaN         265.0                       NaN   
2                                 NaN         117.0                       NaN   
3                                 NaN         120.0                       NaN   
4                                 NaN          90.0                     767.0   
...                               ...           ...                       ...   
35031                           283.0           NaN                       NaN   
35032                           287.0           NaN                    1217.0   
35033                           266.0           NaN                       NaN   
35034                           302.0           NaN                       NaN   
35035  

##  Je splite le dataframe 2019 pour traiter les données horaires uniquement.

In [15]:
data_2019_cleaned = data_2019.loc[:,['Time','O\'Connell St Outside Clerys', 'Dawson Street Replacement', 'Westmoreland Street East', 'Dawson Street',\
'Liffey Street', 'Westmoreland Street West', 'Grafton Street']]

In [16]:
data_2019_cleaned

Time  O'Connell St Outside Clerys  \
0      01-01-2019 00:00:00                       1914.0   
1      01-01-2019 00:15:00                          NaN   
2      01-01-2019 00:30:00                          NaN   
3      01-01-2019 00:45:00                          NaN   
4      01-01-2019 01:00:00                        885.0   
...                    ...                          ...   
35031  31-12-2019 22:45:00                          NaN   
35032  31-12-2019 23:00:00                       1967.0   
35033  31-12-2019 23:15:00                          NaN   
35034  31-12-2019 23:30:00                          NaN   
35035  31-12-2019 23:45:00                          NaN   

       Dawson Street Replacement  Westmoreland Street East  Dawson Street  \
0                            NaN                    1670.0          255.0   
1                            NaN                       NaN            NaN   
2                            NaN                       NaN            NaN   
3                            NaN                       NaN            NaN   
4                            NaN                     767.0          209.0   
...                          ...                       ...            ...   
35031                        6.0                       NaN            NaN   
35032                        1.0                    1217.0            NaN   
35033                        4.0                       NaN            NaN   
35034                        5.0                       NaN            NaN   
35035                        1.0                       NaN            NaN   

       Liffey Street  Westmoreland Street West  Grafton Street  
0              881.0                    1988.0           140.0  
1                NaN                       NaN             NaN  
2                NaN                       NaN             NaN  
3                NaN                       NaN             NaN  
4              723.0                    1270.0           215.0  
...              ...                       ...             ...  
35031            NaN                       NaN             NaN  
35032          936.0                    2898.0          1220.0  
35033            NaN                       NaN             NaN  
35034            NaN                       NaN             NaN  
35035            NaN                       NaN             NaN  

[35036 rows x 8 columns]

### Concaténation de la colonne Dawson Street pour un capteur remplacé

In [17]:
data_2019_cleaned['Dawson total']=\
pd.concat([data_2019_cleaned['Dawson Street'].iloc[:30000],\
           data_2019_cleaned['Dawson Street Replacement'].iloc[30000:]])

In [18]:
data_2019_cleaned = data_2019_cleaned.drop(columns=['Dawson Street','Dawson Street Replacement'])

In [465]:
# Je renomme Dawson Street
data_2019_cleaned = data_2019_cleaned.rename(columns = {'Dawson total' : 'Dawson Street'})

### On ne garde que les comptage horaires

In [19]:
for i in range(data_2019_cleaned.shape[0]):
    if i%4 != 0 :
        data_2019_cleaned = data_2019_cleaned.drop(index = i)

In [21]:
data_2019_cleaned = data_2019_cleaned.reset_index(drop=True)

In [22]:
data_2019_cleaned.info()

<class 'pandas.core.frame.DataFrame'>
RangeIndex: 8759 entries, 0 to 8758
Data columns (total 7 columns):
 #   Column                       Non-Null Count  Dtype  
---  ------                       --------------  -----  
 0   Time                         8759 non-null   object 
 1   O'Connell St Outside Clerys  8759 non-null   float64
 2   Westmoreland Street East     8759 non-null   float64
 3   Liffey Street                8759 non-null   float64
 4   Westmoreland Street West     8759 non-null   float64
 5   Grafton Street               8759 non-null   float64
 6   Dawson total                 7607 non-null   float64
dtypes: float64(6), object(1)
memory usage: 479.1+ KB


Il reste toujours à combler la colonne Dawson Street

## Je passe au reste du dataframe 2019

In [124]:
data_2019_test = data_2019.drop(columns=['Time','O\'Connell St Outside Clerys', 'Dawson Street Replacement', 'Westmoreland Street East', 'Dawson Street',\
'Liffey Street', 'Westmoreland Street West', 'Grafton Street'])

In [125]:
data_2019_test

O'Connell St Outside Pennys  Mary Street  Capel Street  Aston Quay  \
0                              244           20            16         NaN   
1                              454           24            62         NaN   
2                              391           69            99         NaN   
3                              415           50            61         NaN   
4                              319           24            47         NaN   
...                            ...          ...           ...         ...   
35031                          389           13            38       797.0   
35032                          324           18            39       914.0   
35033                          306           19            69       724.0   
35034                          249           11            69       971.0   
35035                          302           14            52       513.0   

       Grafton Street @ CompuB  Talbot Street North  \
0                          NaN                  NaN   
1                          NaN                  NaN   
2                          NaN                  NaN   
3                          NaN                  NaN   
4                          NaN                  NaN   
...                        ...                  ...   
35031                    782.0                 52.0   
35032                    743.0                 48.0   
35033                    821.0                 38.0   
35034                    795.0                 69.0   
35035                    556.0                 47.0   

       Doilier Street, Burgh Quay  Dame Street  Talbot Street South  \
0                             NaN          NaN                  NaN   
1                             NaN          NaN                  NaN   
2                             NaN          NaN                  NaN   
3                             NaN          NaN                  NaN   
4                             NaN          NaN                  NaN   
...                           ...          ...                  ...   
35031                       173.0        226.0                445.0   
35032                       220.0        274.0                353.0   
35033                       241.0        274.0                277.0   
35034                       357.0        383.0                348.0   
35035                       339.0        288.0                584.0   

       O'Connell St, Parnell St @ AIB  \
0                                 NaN   
1                                 NaN   
2                                 NaN   
3                                 NaN   
4                                 NaN   
...                               ...   
35031                           165.0   
35032                           124.0   
35033                           141.0   
35034                           110.0   
35035                           137.0   

       Grafton Street / Nassau Street / Suffolk Street  \
0                                                  NaN   
1                                                  NaN   
2                                                  NaN   
3                                                  NaN   
4                                                  NaN   
...                                                ...   
35031                                            144.0   
35032                                            110.0   
35033                                            170.0   
35034                                            116.0   
35035                                            135.0   

       College Green, Bank Of Ireland  Henry Street  Bachelors Walk  \
0                                 NaN          95.0             NaN   
1                                 NaN         265.0             NaN   
2                                 NaN         117.0             NaN   
3                                 NaN         120.0             NaN   
4                      

In [117]:
data_2019_rest_recept = data_2019_rest

In [118]:
for column in data_2019_rest:
    for i in range(1,data_2019_rest.shape[0]):
        if i%4 == 0 :
            data_2019_rest_recept.loc[i, column] = sum(data_2019_rest.loc[i-3:i, column])

### J'enlève tout ce qui n'est pas un comptage horaire.

In [119]:
for i in range(test_sum.shape[0]):
    if i%4 != 0 :
        data_2019_rest_recept = data_2019_rest_recept.drop(index = i)
    data_2019_rest_recept = data_2019_rest_recept

In [121]:
data_2019_rest_recept = data_2019_rest_recept.reset_index(drop=True)

In [135]:
data_2019_rest_recept.info()

<class 'pandas.core.frame.DataFrame'>
RangeIndex: 8759 entries, 0 to 8758
Data columns (total 16 columns):
 #   Column                                           Non-Null Count  Dtype  
---  ------                                           --------------  -----  
 0   O'Connell St Outside Pennys                      8759 non-null   int64  
 1   Mary Street                                      8759 non-null   int64  
 2   Capel Street                                     8759 non-null   int64  
 3   Aston Quay                                       7340 non-null   float64
 4   Grafton Street @ CompuB                          7390 non-null   float64
 5   Talbot Street North                              7346 non-null   float64
 6   Doilier Street, Burgh Quay                       7387 non-null   float64
 7   Dame Street                                      7390 non-null   float64
 8   Talbot Street South                              7382 non-null   float64
 9   O'Connell St, Parnell St @ AIB

## Je reforme le dataframe 2019

In [215]:
df2019_cleaned = pd.concat([data_2019_cleaned,data_2019_rest_recept], axis=1)

In [152]:
# Il manque un comptage à 01:00:00 le 31-03-2019

df2019_cleaned.iloc[2136:2138]

Time  O'Connell St Outside Clerys  \
2136  31-03-2019 00:00:00                        495.0   
2137  31-03-2019 02:00:00                        343.0   

      Westmoreland Street East  Liffey Street  Westmoreland Street West  \
2136                     431.0          434.0                     841.0   
2137                     269.0          354.0                     736.0   

      Grafton Street  Dawson total  O'Connell St Outside Pennys  Mary Street  \
2136           941.0         333.0                          795           78   
2137           931.0         232.0                          507           50   

      Capel Street  ...  Doilier Street, Burgh Quay  Dame Street  \
2136           196  ...                      1568.0        737.0   
2137           112  ...                      1356.0        539.0   

      Talbot Street South  O'Connell St, Parnell St @ AIB  \
2136               3004.0                           389.0   
2137               3070.0                           239.0   

      Grafton Street / Nassau Street / Suffolk Street  \
2136                                            353.0   
2137                                            239.0   

      College Green, Bank Of Ireland  Henry Street  Bachelors Walk  \
2136                          1112.0         299.0           693.0   
2137                           780.0         179.0           487.0   

      College Green @ Church Lane  College Green - Dame St Side  
2136                        681.0                         518.0  
2137                        404.0                         348.0  

[2 rows x 23 columns]

### Je remplis les valeurs manquantes

In [418]:
df2019_cleaned

Time  O'Connell St Outside Clerys  \
0     01-01-2019 00:00:00                       1914.0   
1     01-01-2019 01:00:00                        885.0   
2     01-01-2019 02:00:00                        984.0   
3     01-01-2019 03:00:00                        935.0   
4     01-01-2019 04:00:00                        390.0   
...                   ...                          ...   
8754  31-12-2019 19:00:00                       1750.0   
8755  31-12-2019 20:00:00                       1644.0   
8756  31-12-2019 21:00:00                       1554.0   
8757  31-12-2019 22:00:00                       1972.0   
8758  31-12-2019 23:00:00                       1967.0   

      Westmoreland Street East  Liffey Street  Westmoreland Street West  \
0                       1670.0          881.0                    1988.0   
1                        767.0          723.0                    1270.0   
2                        642.0         1010.0                    1589.0   
3                        582.0          772.0                    1534.0   
4                        143.0          197.0                     610.0   
...                        ...            ...                       ...   
8754                     763.0          910.0                    1929.0   
8755                     727.0          989.0                    1984.0   
8756                     752.0          779.0                    2120.0   
8757                     997.0          823.0                    2461.0   
8758                    1217.0          936.0                    2898.0   

      Grafton Street  Dawson total  O'Connell St Outside Pennys  Mary Street  \
0              140.0         255.0                          244           20   
1              215.0         209.0                         1579          167   
2              210.0         193.0                         1164           85   
3              204.0         178.0                         1204           79   
4               88.0          35.0                         1308           52   
...              ...           ...                          ...          ...   
8754          1190.0          43.0                         1634          236   
8755          1239.0          61.0                         1571          226   
8756          1314.0           6.0                         1432          131   
8757          1315.0           9.0                         1415           86   
8758          1220.0           1.0                         1425           64   

      Capel Street  ...  Doilier Street, Burgh Quay  Dame Street  \
0               16  ...                         NaN          NaN   
1              269  ...                         NaN          NaN   
2              149  ...                         NaN          NaN   
3              125  ...                         NaN          NaN   
4              179  ...                         NaN          NaN   
...            ...  ...                         ...          ...   
8754           331  ...                       278.0       1070.0   
8755           324  ...                       352.0       1118.0   
8756           271  ...                       450.0       1194.0   
8757           218  ...                       492.0       1335.0   
8758           197  ...                       745.0       1123.0   

      Talbot Street South  O'Connell St, Parnell St @ AIB  \
0                     NaN                             NaN   
1                     NaN                             NaN   
2                     NaN                             NaN   
3                     NaN                             NaN   
4                     NaN                             NaN   
...                   ...                             ...   
8754                811.0                           916.0   
8755                740.0                           823.0   
8756                990.0                           628.0   
8757               1434.0          

In [420]:
df_test = df2019_cleaned.copy()

In [421]:
# Je reprendrai cette stratégie pour les différentes colonnes des dataset suivants.
df_test['Time'] = pd.to_datetime(df_test['Time'])

In [422]:
# J'ajoute une colonne pour rassembler tous les comptages par heure

df_test['hour_cycle'] = 0
for i in range(df_test.shape[0]):
    df_test['hour_cycle'][i] = df_test['Time'][i].hour

/tmp/ipykernel_1516/8810359.py:3: SettingWithCopyWarning: 
A value is trying to be set on a copy of a slice from a DataFrame

See the caveats in the documentation: https://pandas.pydata.org/pandas-docs/stable/user_guide/indexing.html#returning-a-view-versus-a-copy
  df_test['hour_cycle'][i] = df_test['Time'][i].hour


In [423]:
df_test

Time  O'Connell St Outside Clerys  \
0    2019-01-01 00:00:00                       1914.0   
1    2019-01-01 01:00:00                        885.0   
2    2019-01-01 02:00:00                        984.0   
3    2019-01-01 03:00:00                        935.0   
4    2019-01-01 04:00:00                        390.0   
...                  ...                          ...   
8754 2019-12-31 19:00:00                       1750.0   
8755 2019-12-31 20:00:00                       1644.0   
8756 2019-12-31 21:00:00                       1554.0   
8757 2019-12-31 22:00:00                       1972.0   
8758 2019-12-31 23:00:00                       1967.0   

      Westmoreland Street East  Liffey Street  Westmoreland Street West  \
0                       1670.0          881.0                    1988.0   
1                        767.0          723.0                    1270.0   
2                        642.0         1010.0                    1589.0   
3                        582.0          772.0                    1534.0   
4                        143.0          197.0                     610.0   
...                        ...            ...                       ...   
8754                     763.0          910.0                    1929.0   
8755                     727.0          989.0                    1984.0   
8756                     752.0          779.0                    2120.0   
8757                     997.0          823.0                    2461.0   
8758                    1217.0          936.0                    2898.0   

      Grafton Street  Dawson total  O'Connell St Outside Pennys  Mary Street  \
0              140.0         255.0                          244           20   
1              215.0         209.0                         1579          167   
2              210.0         193.0                         1164           85   
3              204.0         178.0                         1204           79   
4               88.0          35.0                         1308           52   
...              ...           ...                          ...          ...   
8754          1190.0          43.0                         1634          236   
8755          1239.0          61.0                         1571          226   
8756          1314.0           6.0                         1432          131   
8757          1315.0           9.0                         1415           86   
8758          1220.0           1.0                         1425           64   

      Capel Street  ...  Dame Street  Talbot Street South  \
0               16  ...          NaN                  NaN   
1              269  ...          NaN                  NaN   
2              149  ...          NaN                  NaN   
3              125  ...          NaN                  NaN   
4              179  ...          NaN                  NaN   
...            ...  ...          ...                  ...   
8754           331  ...       1070.0                811.0   
8755           324  ...       1118.0                740.0   
8756           271  ...       1194.0                990.0   
8757           218  ...       1335.0               1434.0   
8758           197  ...       1123.0               1575.0   

      O'Connell St, Parnell St @ AIB  \
0                                NaN   
1                                NaN   
2                                NaN   
3                                NaN   
4                                NaN   
...                              ...   
8754                           916.0   
8755                           823.0   
8756                           628.0   
8757                           509.0   
8758                           532.0   

      Grafton Street / Nassau Street / Suffolk Street  \
0                                                 NaN   
1                                                 NaN   
2                                                 NaN   
3                              

In [425]:
df_test['hour_cycle'].value_counts()

0     365
13    365
22    365
21    365
20    365
19    365
18    365
17    365
16    365
15    365
14    365
12    365
11    365
10    365
9     365
8     365
7     365
6     365
5     365
4     365
3     365
2     365
23    365
1     364
Name: hour_cycle, dtype: int64

In [429]:
# Je sors toutes les colonnes avec des valeurs manquantes
df_test_2 = df_test[['Dame Street', 'Aston Quay','Grafton Street @ CompuB','Talbot Street North',\
                     'Doilier Street, Burgh Quay','Dame Street','Talbot Street South',\
                     'O\'Connell St, Parnell St @ AIB','Grafton Street / Nassau Street / Suffolk Street',\
                     'College Green, Bank Of Ireland','Henry Street','Bachelors Walk',\
                     'College Green @ Church Lane','College Green - Dame St Side' ,'hour_cycle']]

In [430]:
# je les rassemble par heure, en virant les Nan
df_test_2_count = df_test_2.dropna().groupby(by="hour_cycle").agg('median').reset_index(drop=False)

In [431]:
#J'obtiens la distribution médiane par heure pour chaque rue
df_test_2_count

hour_cycle  Dame Street  Aston Quay  Grafton Street @ CompuB  \
0            0        363.0       522.0                    890.0   
1            1        182.0       234.5                    434.0   
2            2        123.0       143.0                    291.0   
3            3        108.0       116.0                    254.0   
4            4         85.0        87.0                    216.0   
5            5         48.0        48.0                    141.0   
6            6         82.0        84.0                    245.0   
7            7        393.0       371.0                   1007.0   
8            8        617.0       619.0                   1732.0   
9            9        996.0       914.0                   2387.0   
10          10        886.0       707.0                   2073.0   
11          11        962.0       714.0                   2149.0   
12          12       1111.0       798.0                   2517.0   
13          13       1417.0      1025.0                   3038.0   
14          14       1527.0      1111.0                   3177.0   
15          15       1428.0      1016.0                   3062.0   
16          16       1435.5      1120.5                   3128.0   
17          17       1529.5      1304.0                   3323.0   
18          18       1631.5      1403.5                   3478.5   
19          19       1295.0      1306.0                   3144.0   
20          20       1038.0      1142.0                   2552.0   
21          21        820.0      1008.0                   2025.0   
22          22        719.0       915.0                   1650.0   
23          23        674.0       834.0                   1452.0   

    Talbot Street North  Doilier Street, Burgh Quay  Dame Street  \
0                  92.0                       442.0        363.0   
1                  47.5                       151.0        182.0   
2                  29.0                        96.0        123.0   
3                  21.0                       195.0        108.0   
4                  18.0                        40.0         85.0   
5                  17.0                        19.0         48.0   
6                  27.0                        67.0         82.0   
7                  85.0                       421.0        393.0   
8                 186.0                       722.0        617.0   
9                 252.0                       943.0        996.0   
10                203.0                       793.0        886.0   
11                195.0                       756.0        962.0   
12                226.0                       771.0       1111.0   
13                306.0                       935.0       1417.0   
14                282.0                       954.0       1527.0   
15                238.0                       827.0       1428.0   
16                233.5                       818.5       1435.5   
17                271.5                       885.5       1529.5   
18                280.0                       953.5       1631.5   
19                233.0                       824.0       1295.0   
20                146.0                       739.0       1038.0   
21                127.0                       648.0        820.0   
22                110.0                       627.0        719.0   
23                105.0                       629.0        674.0   

    Talbot Street South  O'Connell St, Parnell St @ AIB  \
0                1903.0                           244.0   
1                2556.5                           118.0   
2                2445.0                            69.0   
3                2003.0                            60.0   
4                1606.0                            44.0   
5                1240.0                            34.0   
6                1179.0                            79.0   
7                1824.0                           237.0   
8                2241.0                           415.0   
9           

Liste des rues avec des valeurs manquantes :

Aston Quay                                         1419 <br>
Grafton Street @ CompuB                            1369<br>
Talbot Street North                                1413<br>
Doilier Street, Burgh Quay                         1372<br>
Dame Street                                        1369<br>
Talbot Street South                                1377<br>
O'Connell St, Parnell St @ AIB                     1851<br>
Grafton Street / Nassau Street / Suffolk Street    1369<br>
College Green, Bank Of Ireland                     1369<br>
Henry Street                                       1656<br>
Bachelors Walk                                     1522<br>
College Green @ Church Lane                        1516<br>
College Green - Dame St Side                       1516<br>

In [321]:
df_test.describe()

O'Connell St Outside Clerys  Westmoreland Street East  Liffey Street  \
count                  8759.000000               8759.000000    8759.000000   
mean                   1021.754082                543.128439     771.072040   
std                     717.378340                399.249513     689.053236   
min                       0.000000                  0.000000       0.000000   
25%                     311.500000                160.500000     133.000000   
50%                    1049.000000                532.000000     620.000000   
75%                    1609.000000                833.000000    1288.000000   
max                    3798.000000               2852.000000    4388.000000   

       Westmoreland Street West  Grafton Street  Dawson total  \
count               8759.000000     8759.000000   7607.000000   
mean                 998.951935     1224.335883    343.045090   
std                  629.605104     1113.288473    345.033782   
min                    0.000000        0.000000      0.000000   
25%                  447.000000      288.000000     33.500000   
50%                 1017.000000      827.000000    224.000000   
75%                 1409.000000     1959.500000    591.000000   
max                 3280.000000     5427.000000   2047.000000   

       O'Connell St Outside Pennys  Mary Street  Capel Street   Aston Quay  \
count                  8759.000000  8759.000000   8759.000000  7340.000000   
mean                   1407.714922   236.107204    256.269551   965.179155   
std                    1157.034060   259.109360    275.685225   754.700161   
min                       0.000000     0.000000      0.000000     8.000000   
25%                     343.000000    24.000000     56.000000   425.000000   
50%                    1102.000000   119.000000    234.000000   865.500000   
75%                    2397.500000   399.000000    407.000000  1236.250000   
max                    4803.000000  1257.000000   5900.000000  4637.000000   

       ...  Dame Street  Talbot Street South  O'Connell St, Parnell St @ AIB  \
count  ...  7390.000000          7382.000000                     6908.000000   
mean   ...   799.207984          2273.700352                      723.681529   
std    ...   556.455263          2948.484265                      545.835246   
min    ...     0.000000             4.000000                        0.000000   
25%    ...   266.000000           956.000000                      163.750000   
50%    ...   786.000000          1451.000000                      709.000000   
75%    ...  1228.000000          2150.000000                     1229.000000   
max    ...  2395.000000         17414.000000                     2043.000000   

       Grafton Street / Nassau Street / Suffolk Street  \
count                                      7390.000000   
mean                                        459.918539   
std                                         383.601685   
min                                           0.000000   
25%                                         114.000000   
50%                                         371.500000   
75%                                         748.000000   
max                                        2640.000000   

       College Green, Bank Of Ireland  Henry Street  Bachelors Walk  \
count                     7390.000000   7103.000000     7237.000000   
mean                       791.244790   1797.965648      786.472295   
std                        614.331854   1750.289867     1326.440921   
min                          0.000000      5.000000        0.000000   
25%                        243.000000    149.000000      180.000000   
50%                        670.500000   1054.000000      540.000000   
75%                       1224.000000   3417.000000      885.000000   
max                       4233.000000   6774.000000    11345.000000   

       College Green @ Church Lane  College Green - Dame St Side   hour_cycle  
count                  7243.000000        

# Dataset 2020

In [163]:
data_2020 = pd.read_csv('../raw_data/ped_data_2020.csv')

In [168]:
data_2020.shape

(8784, 23)

# Liste des capteurs ajoutés en 2021 (à retirer ?)

2   Baggot st lower/Wilton tce inbound               5079 non-null   float64<br>
 3   Baggot st upper/Mespil rd/Bank <br>

 15  Grand Canal st upp/Clanwilliam place             5078 non-null   float64<br>
 16  Grand Canal st upp/Clanwilliam place/Google<br>

Newcomen Bridge/Charleville mall inbound         5075 non-null   float64<br>
 21  Newcomen Bridge/Charleville mall outbound        5074 non-null   float64<br>
 22  North Wall Quay/Samuel Beckett bridge East       5087 non-null   float64<br>
 23  North Wall Quay/Samuel Beckett bridge West <br>

27  Phibsborough Rd/Enniskerry Road                  5087 non-null   float64<br>
 28  Phibsborough Rd/Munster St                       798 non-null    float64<br>
 29  Richmond st south/Portabello Harbour inbound     5079 non-null   float64<br>
 30  Richmond st south/Portabello Harbour outbound<br>